lib/common.js

```diff
@@ -3,29 +3,35 @@ const fs = require('fs');
 
 module.exports = {
     push_test_state_to_results(tests_results, test_key, test_comment, test_evidences, test, test_type, test_info_steps = null, scenario = null) {
-        if (test.state === "passed") {
+        // Check if actual test have retries to push only the last result when it fails
+        const test_have_retries = test._retries !== -1;
+        const is_retry_test = test?._retriedTest !== undefined;
+        const is_last_retry = test?._retries === test?._currentRetry;
+        const test_state = is_retry_test ? test._retriedTest.state : test.state;
+
+        if (test_state === "passed") {
             tests_results.push({
                 title: test.title,
                 status: "PASSED",
                 comment: "Test passed successfully",
                 test_key: test_key,
                 start: moment(test.startedAt).format(),
                 finish: moment().format(),
-                id: test.id,
+                id: test?.uid,
```

Comment:
Curious why this change?  was it related to this fix?

Reply:
not related, just for testing purpose 👍

Reply:
Done on [229dd1f](https://github.com/saadichouaib/codeceptjs-xray-cloud-helper/pull/10/commits/229dd1fb14bc55b14e912cf93c5842da41f7952d) 👍